src/ci_tools.ts

```diff
@@ -1,33 +1,74 @@
-import { run as runAutoPublishIfApplicable } from './commands/auto-publish-if-applicable';
+import chalk from 'chalk';
+
+import { run as runAutoPublishIfApplicable } from './commands/internal/auto-publish-if-applicable';
+import { run as runCommitAndTagVersion } from './commands/commit-and-tag-version';
+import { run as runCreateChangelog } from './commands/internal/create-changelog';
 import { run as runNpmInstallOnly } from './commands/npm-install-only';
-import { run as runCreateChangelog } from './commands/create-changelog';
+import { run as runPrepareVersion } from './commands/prepare-version';
+import { run as runSetupGitAndNpmConnections } from './commands/internal/setup-git-and-npm-connections';
 import { run as runUpdateGithubRelease } from './commands/update-github-release';
-import { run as runIncrementVersion } from './commands/increment-version';
-import { run as runSetupGitAndNpm } from './commands/setup-git-and-npm';
+
+import { getGitBranch } from './git/git';
+import { PRIMARY_BRANCHES } from './versions/increment_version';
 
 const COMMAND_HANDLERS = {
+  'commit-and-tag-version': runCommitAndTagVersion,
+  'prepare-version': runPrepareVersion,
+  'npm-install-only': runNpmInstallOnly,
+  'update-github-release': runUpdateGithubRelease
+};
+
+// Internal commands are only used to develop ci_tools and are not intended for public consumption.
+const INTERNAL_COMMAND_HANDLERS = {
   'auto-publish-if-applicable': runAutoPublishIfApplicable,
   'create-changelog': runCreateChangelog,
-  'increment-version': runIncrementVersion,
-  'npm-install-only': runNpmInstallOnly,
-  'update-github-release': runUpdateGithubRelease,
-  'setup-git-and-npm': runSetupGitAndNpm
+  'setup-git-and-npm-connections': runSetupGitAndNpmConnections
 };
 
-const [, , ...args] = process.argv;
+function run(argv: string[]): void {
+  const [, , ...args] = argv;
+
+  if (args.length === 0) {
+    console.log('Usage: ci_tools <COMMAND>');
+    console.log('');
+    console.log('COMMAND can be any of:');
+    Object.keys(COMMAND_HANDLERS).forEach((commandName: string): void => console.log(`  ${commandName}`));
+    process.exit(1);
+  }
+  const [commandName, ...restArgs] = args;
 
-if (args.length === 0) {
-  console.log('Usage: ci_tools <COMMAND>');
-  console.log('');
-  console.log('COMMAND can be any of:');
-  Object.keys(COMMAND_HANDLERS).forEach((commandName: string): void => console.log(`  ${commandName}`));
-  process.exit(1);
+  const foundCommand = COMMAND_HANDLERS[commandName] || INTERNAL_COMMAND_HANDLERS[commandName];
+
+  if (foundCommand == null) {
```

Comment:
I prefer this:

```suggestion
  if (!foundCommand) {
```

Because this will cover `null` and `undefined`.

Otherwise, please use:

```suggestion
  if (foundCommand === undefined) {
```

Because if something doesn't exist in NodeJS, it will be declared `undefined` by default, not null.

Reply:
`== null` deckt `null` und `undefined` ab ([hier ist eine gute Referenz](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/null#Difference_between_null_and_undefined), wie Vergleiche in dem Fall funktionieren). Praktisch gesprochen ist `== null` der eine Fall, in dem man Type Coercion immer haben will und deshalb `==` nutzen sollte anstelle von `===`.

Die Verwendung von `===` bei `null` oder `undefined` ist nachteilig (um nicht zu sagen, grob fahrlässig), da die Unterscheidung im eigenen Code immer subjektiv (und der Unterschied ohnehin akademischer Natur) ist und alle Annahmen `null` und nichtig sind, sobald mit Code von anderen Leuten arbeiten muss 😉